### ISSUE TITLE
_XSERVTransSocketWritev (Xtranssock.c:2978) -> writev(vector[...]) points to uninitialised byte(s)

### ISSUE THREAD

**Issue by sunweaver** [ArcticaProject/nx-libs#324]
Regularly on nxagent startup, I see the below issue when launching nxagent in valgrind.

The error output appears when starting a desktop session (tested with RazorQt and XFCE4 from Debian 9) in nxagent.

```
==10999== Syscall param writev(vector[...]) points to uninitialised byte(s)
==10999==    at 0x77E1D40: __writev_nocancel (syscall-template.S:84)
==10999==    by 0x1A86B6: _XSERVTransSocketWritev (Xtranssock.c:2978)
==10999==    by 0x19EF04: FlushClient (io.c:1052)
==10999==    by 0x19F406: FlushAllOutput (io.c:800)
==10999==    by 0x166D74: Dispatch (NXdispatch.c:537)
==10999==    by 0x14A762: main (main.c:433)
==10999==  Address 0xe86acb1 is 1 bytes inside a block of size 4,096 alloc'd
==10999==    at 0x4C2BBAF: malloc (vg_replace_malloc.c:299)
==10999==    by 0x19FD5C: AllocateOutputBuffer (io.c:1192)
==10999==    by 0x1A0034: WriteToClient (io.c:911)
==10999==    by 0x16549C: SendConnSetup (dispatch.c:3892)
==10999==    by 0x166E7E: Dispatch (NXdispatch.c:491)
==10999==    by 0x14A762: main (main.c:433)
```

**Comment by sunweaver** [comment]
running valgrind with --track-origins=yes reveals this:

```
==13181==  Uninitialised value was created by a stack allocation
==13181==    at 0x265D30: __glXQueryVersion (glxcmds.c:775)
```

**Comment by uli42** [comment]
So maybe Mesa is the problem. Do you see these when disabling the GLX extension?

**Comment by uli42** [comment]
Can you please go back to a commit prior to the xtrans upgrade and re-check? I have tried that for #306 but got stuck in strange compilation issues and then ran out of time...

**Comment by uli42** [comment]
I have identified teh cause of lots of valgrind messages like this. And I have fixed a lot already. Most of them are structures that are going to be sent to some client (events and replies). The fixes are basically the same as in commit d088698324d5e71cb93ccd429f084729ba07872c:
```
Author: Peter Åstrand <astrand@cendio.se>
Date:   Fri Feb 13 10:23:28 2009 +0100

    Backport: xserver: Avoid sending uninitialized padding data over the network
    
    Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
```

Preparing a big patch...
